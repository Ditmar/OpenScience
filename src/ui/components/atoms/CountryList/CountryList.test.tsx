// import { render, screen, fireEvent } from '@testing-library/react';
// import { vi } from 'vitest';
// import { CountryListDropdown } from './CountryList';

// const mockCountries = [
//   { code: 'ES', name: 'Spain', flag: 'ðŸ‡ªðŸ‡¸', dialCode: '+34' },
//   { code: 'US', name: 'United States', flag: 'ðŸ‡ºðŸ‡¸', dialCode: '+1' },
//   { code: 'FR', name: 'France', flag: 'ðŸ‡«ðŸ‡·', dialCode: '+33' },
// ];

// describe('CountryList Component', () => {
//   it('renders correctly with props', () => {
//     render(
//       <CountryListDropdown
//         countries={mockCountries}
//         selectedCountry="ES"
//         isOpen
//         onClose={() => {}}
//         onSelectCountry={() => {}}
//       />,
//     );

//     expect(screen.getByText('Spain')).toBeInTheDocument();
//     expect(screen.getByText('United States')).toBeInTheDocument();
//     expect(screen.getByText('France')).toBeInTheDocument();
//   });

//   it('selects a country on click', () => {
//     const onSelect = vi.fn();
//     render(
//       <CountryListDropdown
//         countries={mockCountries}
//         selectedCountry=""
//         isOpen
//         onClose={() => {}}
//         onSelectCountry={onSelect}
//       />,
//     );

//     fireEvent.click(screen.getByText('France'));

//     expect(onSelect).toHaveBeenCalledWith('FR');
//   });

//   it('allows scrolling', () => {
//     render(
//       <CountryListDropdown
//         countries={mockCountries}
//         selectedCountry=""
//         isOpen
//         onClose={() => {}}
//         onSelectCountry={() => {}}
//       />,
//     );

//     const dropdown = screen.getByRole('list');
//     fireEvent.scroll(dropdown, { target: { scrollTop: 50 } });

//     expect(dropdown.scrollTop).toBe(50);
//   });

//   it('selects a country using keyboard navigation', () => {
//     const onSelect = vi.fn();
//     render(
//       <CountryListDropdown
//         countries={mockCountries}
//         selectedCountry=""
//         isOpen
//         onClose={() => {}}
//         onSelectCountry={onSelect}
//       />,
//     );

//     fireEvent.keyDown(screen.getByText('Spain'), { key: 'ArrowDown' });
//     fireEvent.keyDown(screen.getByText('United States'), { key: 'Enter' });

//     expect(onSelect).toHaveBeenCalledWith('US');
//   });
// });
