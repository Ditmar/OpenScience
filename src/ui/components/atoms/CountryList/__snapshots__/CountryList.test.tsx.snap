// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CountryList Component > matches snapshot 1`] = `
<div>
  <div
    class="_country-list__container_0abd6d"
  >
    <button
      class="_country-list__select-button_0abd6d"
    >
      US
    </button>
    <div
      class="_country-list__dropdown_0abd6d _scrollbar_0abd6d"
    >
      <div
        class="_country-list__list_0abd6d"
      >
        <div
          class="_country-list__item_0abd6d "
          role="button"
          tabindex="0"
        >
          <img
            alt="Spain"
            class="_country-list__flag_0abd6d"
            src="ðŸ‡ªðŸ‡¸"
          />
          <span
            class="_country-list__name_0abd6d"
          >
            Spain
          </span>
          <span
            class="_country-list__dial-code_0abd6d"
          >
            (
            +34
            )
          </span>
        </div>
        <div
          class="_country-list__item_0abd6d "
          role="button"
          tabindex="0"
        >
          <img
            alt="United States"
            class="_country-list__flag_0abd6d"
            src="ðŸ‡ºðŸ‡¸"
          />
          <span
            class="_country-list__name_0abd6d"
          >
            United States
          </span>
          <span
            class="_country-list__dial-code_0abd6d"
          >
            (
            +1
            )
          </span>
        </div>
        <div
          class="_country-list__item_0abd6d "
          role="button"
          tabindex="0"
        >
          <img
            alt="France"
            class="_country-list__flag_0abd6d"
            src="ðŸ‡«ðŸ‡·"
          />
          <span
            class="_country-list__name_0abd6d"
          >
            France
          </span>
          <span
            class="_country-list__dial-code_0abd6d"
          >
            (
            +33
            )
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
